lib/resolvers/pil.js

```diff
@@ -33,20 +34,25 @@ module.exports = ({ models }) => ({ action, data, id }, transaction) => {
       }
     }
 
-    return Promise.resolve()
-      .then(() => generateLicenceNumber(PIL, transaction, 'pil'))
-      .then(licenceNumber => {
-        return PIL.query(transaction).findById(id)
-          .then(pil => pil.$query(transaction).patchAndFetch({
-            status: 'active',
-            issueDate: new Date().toISOString(),
-            licenceNumber: pil.licenceNumber || licenceNumber,
-            species: data.species,
-            procedures: data.procedures,
-            notesCatD: data.notesCatD,
-            notesCatF: data.notesCatF
-          }));
-      });
+    const licenceNumber = await generateLicenceNumber(PIL, transaction, 'pil');
+    const pil = await PIL.query(transaction).findById(id);
+    const profile = await Profile.query(transaction).findById(changedBy);
```

Comment:
Are we _sure_ this is the person who initiated the request and not the person who resolved the task?

Context: I am not sure.

Reply:
hmm, I am not sure either now, I thought it was only set on create task and not on status changes, I'll check.

Reply:
Both will be available _somewhere_ on the event object, so just need to make sure we're getting the right one.